R/get_data.R

```diff
@@ -59,47 +56,84 @@ bcdc_get_data.bcdc_record <- function(record, format = NULL, resource = NULL, ..
 }
 
 #' @export
-bcdc_get_data.character <- function(record, format = NULL, resource = NULL, ...) {
+bcdc_get_data.character <- function(record, resource = NULL, ...) {
   x <- slug_from_url(record)
 
-  if(is.null(format)){
-    query <- bcdc_query_geodata(record = record, ...)
-    return(collect(query))
+  record <- bcdc_get_record(x)
+
+  resource_df <- dplyr::tibble(
+    name = purrr::map_chr(record$resources, "name"),
+    url = purrr::map_chr(record$resources, "url"),
+    id = purrr::map_chr(record$resources, "id"),
+    format = purrr::map_chr(record$resources, "format"),
+    ext = tools::file_ext(url),
+    location = tolower(
+      gsub("\\s", "", purrr::map_chr(record$resources, "resource_storage_location"))
+      )
+  )
+
+  wms_resource_id <- resource_df$id[resource_df$format == "wms"]
+
+
+  ## wms record; resource supplied
+  if(identical(wms_resource_id, resource)){
+      query <- bcdc_query_geodata(record = x, ...)
+      return(collect(query))
   }
 
-  if(!all(format %in% formats_supported())){
-    stop(paste0("The ", format, " extension is not currently supported by bcdata"),
-         call. = FALSE)
+  ## wms record; no aux resources
+  if(!is_emptyish(wms_resource_id) && all(unique(resource_df$location) %in% c("bcgwdatastore","bcgeographicwarehouse"))){
```

Comment:
I think we should completely remove the `"BC Geographic Warehouse Custom Download"` (`location == ""bcgeographicwarehouse"`) from the resource list as that resource is for the fillable web form for the data distribution service, so is not really relevant here (neither the URL nor the `bcdc_get_data()` call works for this resource). This should possible even be done in `bcdc_get_record` so we don't have to deal with it at all.

I then think the (now simpler)`if` statement could be combined with the above chunk with an `|`:
```r
if (identical(wms_resource_id, resource) || 
    (!is_emptyish(wms_resource_id) && all(unique(resource_df$location) == "bcgwdatastore"))) {
      query <- bcdc_query_geodata(record = x, ...)
      return(collect(query))
}
```

Reply:
I think dealing with it in `bcdc_get_record` is a very good idea. It is actually a new entry and reflects recent changes to the catalogue.